Add documentation of the vulnerability detector tests (#820)
wazuh/wazuh-qa: tests/integration/test_vulnerability_detector/test_feeds/readme.md

# Test feeds

## Objective

> The purpose of testing imported feeds is to check the behavior of the vulnerability detector when importing feeds that
> may have some kind of error.

Reviewer: ```suggestion
The purpose of testing imported feeds is to check the behavior of Vulnerability Detector when importing feeds that may have some kind of error.
```

## General info

|Tier | Number of tests | Time spent |
|:--:|:--:|:--:|
| 1-2 | 1134 | 2h 53m |

## Expected behavior

> The expected behavior is to **discard** those vulnerabilities whose content is malformed making vulnerability detector
> **not fail** for any related reason.

Reviewer: ```suggestion
The expected behavior is to **discard** those vulnerabilities whose content is malformed making Vulnerability Detector **not fail** for it.
```

## Testing

The tests can be classified into:

> - **Generics**: Generic tests that are carried out on the feeds, regardless of the provider.
> - **Providers**: Specific tests of each provider in which the feed is altered and the result is evaluated.

Reviewer: ```suggestion
- **Generics**: tests that are carried out on the feeds, regardless of the provider.
- **Providers**: specific tests of each provider where the feed is modified and the result is evaluated.
```

### Generic tests

> - **[test_download_feeds](../doc/test_feeds/generic/test_download_feeds.md)**: Tests that are based on downloading the
> different feeds (redhat, canonical, debian and NVD) and looking for the message confirming that the feeds have been
> downloaded and imported correctly.

Reviewer: ```suggestion
- **[test_download_feeds](../doc/test_feeds/generic/test_download_feeds.md)**: Tests that download the different feeds (Redhat, Canonical, Debian, and NVD) and check the confirmation message saying the feeds have been correctly downloaded and imported.
```

> - **[test_invalid_type_custom_feeds](../doc/test_feeds/generic/test_invalid_type_custom_feeds.md)**: Tests that are
> based on importing files of several different types (`.mp3`, `.jpg`, `.pdf` ...) as custom feed, and observing the
> response of vulnerability detector.

Reviewer: ```suggestion
- **[test_invalid_type_custom_feeds](../doc/test_feeds/generic/test_invalid_type_custom_feeds.md)**: Tests that importing files of several different types (`.mp3`, `.jpg`, `.pdf` ...) as custom feed, and check the response of Vulnerability Detector.
```

> - **[test_validate_feed_content](../doc/test_feeds/generic/test_validate_feed_content.md)**: Tests that are based on
> downloading the feeds of all providers and verifying that the format of each feed is as expected and that the
> content of the feed is also `XML` or `JSON` parseable.

Reviewer: ```suggestion
- **[test_validate_feed_content](../doc/test_feeds/generic/test_validate_feed_content.md)**: Tests that download the feeds of all the providers and verify their format is as expected and their content is also `XML` or `JSON` parseable.
```

### Providers tests

> #### Red Hat

Reviewer: ```suggestion
#### RedHat
```

> - **[test_extra_fields_redhat_feed](../doc/test_feeds/redhat/test_extra_fields_redhat_feed.md)**: Set of tests that are
> based on checking the behavior of vulnerability detector when there is any extra field in the feed.

Reviewer: ```suggestion
- **[test_extra_fields_redhat_feed](../doc/test_feeds/redhat/test_extra_fields_redhat_feed.md)**: Set of tests that check the behavior of Vulnerability Detector when there is any extra field in the feed.
```

> - **[test_invalid_syntax_redhat_feed](../doc/test_feeds/redhat/test_invalid_syntax_redhat_feed.md)**: Set of tests that
> are based on checking the behavior of vulnerability detector when the feed has some kind of syntactic error: it doesn't
> have a closing field, it's missing some character ...

Reviewer: ```suggestion
- **[test_invalid_syntax_redhat_feed](../doc/test_feeds/redhat/test_invalid_syntax_redhat_feed.md)**: Set of tests that check the behavior of Vulnerability Detector when the feed has syntactic errors.
```

> - **[test_invalid_values_redhat_feed](../doc/test_feeds/redhat/test_invalid_values_redhat_feed.md)**: Set of tests that
> are based on checking the behavior of vulnerability detector when the value of a field is not correct: invalid type,
> strange characters ...

Reviewer: ```suggestion
- **[test_invalid_values_redhat_feed](../doc/test_feeds/redhat/test_invalid_values_redhat_feed.md)**: Set of tests that check the behavior of Vulnerability Detector when the value of a field is not correct.
```

> - **[test_missing_fields_redhat_feed](../doc/test_feeds/redhat/test_missing_fields_redhat_feed.md)**: Set of tests that
> are based on checking the behavior of vulnerability detector when any field is missing from the feed.

Reviewer: ```suggestion
- **[test_missing_fields_redhat_feed](../doc/test_feeds/redhat/test_missing_fields_redhat_feed.md)**: Set of tests that check the behavior of Vulnerability Detector when any field is missing from the feed.
```

#### Canonical

> - **[test_extra_tags_canonical_feed](../doc/test_feeds/canonical/test_extra_tags_canonical_feed.md)**:  Set of tests
> that are based on checking the behavior of vulnerability detector when there is any extra tag in the feed.

Reviewer: ```suggestion
- **[test_extra_tags_canonical_feed](../doc/test_feeds/canonical/test_extra_tags_canonical_feed.md)**: Set of tests that check the behavior of Vulnerability Detector when there is any extra tag in the feed.
```

> - **[test_invalid_syntax_canonical_feed](../doc/test_feeds/canonical/test_invalid_syntax_canonical_feed.md)**: Set of
> tests that are based on checking the behavior of vulnerability detector when the feed has some kind of syntactic error:
> it doesn't have a closing tag, it's missing some character ...

Reviewer: ```suggestion
- **[test_invalid_syntax_canonical_feed](../doc/test_feeds/canonical/test_invalid_syntax_canonical_feed.md)**: Set of tests that check the behavior of Vulnerability Detector when the feed has some kind of syntactic error.
```

> - **[test_invalid_values_canonical_feed](../doc/test_feeds/canonical/test_invalid_values_canonical_feed.md)**: Set of
> tests that are based on checking the behavior of vulnerability detector when the value of a tag is not correct:
> invalid type, strange characters ...

Reviewer: ```suggestion
- **[test_invalid_values_canonical_feed](../doc/test_feeds/canonical/test_invalid_values_canonical_feed.md)**: Set of tests that check the behavior of Vulnerability Detector when the value of a tag is not correct.
```

> - **[test_missing_tags_canonical_feed](../doc/test_feeds/canonical/test_missing_tags_canonical_feed.md)**: Set of tests
> that are based on checking the behavior of vulnerability detector when any tag is missing from the feed.

Reviewer: ```suggestion
- **[test_missing_tags_canonical_feed](../doc/test_feeds/canonical/test_missing_tags_canonical_feed.md)**: Set of tests that check the behavior of Vulnerability Detector when any tag is missing from the feed.
```

#### Debian

> - **[test_extra_tags_debian_feed](../doc/test_feeds/debian/test_extra_tags_debian_feed.md)**: Set of tests that are
> based on checking the behavior of vulnerability detector when there is any extra tag in the feed.
> 
> - **[test_invalid_syntax_debian_feed](../doc/test_feeds/debian/test_invalid_syntax_debian_feed.md)**: Set of tests that
> are based on checking the behavior of vulnerability detector when the feed has some kind of syntactic error: it doesn't
> have a closing tag, it's missing some character ...
> 
> - **[test_invalid_values_debian_feed](../doc/test_feeds/debian/test_invalid_values_debian_feed.md)**: Set of tests that
> are based on checking the behavior of vulnerability detector when the value of a tag is not correct: invalid type,
> strange characters ...
> 
> - **[test_missing_tags_debian_feed](../doc/test_feeds/debian/test_missing_tags_debian_feed.md)**: Set of tests that are
> based on checking the behavior of vulnerability detector when any tag is missing from the feed.

Reviewer: ```suggestion
- **[test_extra_tags_debian_feed](../doc/test_feeds/debian/test_extra_tags_debian_feed.md)**: Set of tests that check the behavior of Vulnerability Detector when there is any extra tag in the feed.

- **[test_invalid_syntax_debian_feed](../doc/test_feeds/debian/test_invalid_syntax_debian_feed.md)**: Set of tests that checking the behavior of Vulnerability Detector when the feed has some kind of syntactic error.

- **[test_invalid_values_debian_feed](../doc/test_feeds/debian/test_invalid_values_debian_feed.md)**: Set of tests that check the behavior of Vulnerability Detector when the value of a tag is not correct.

- **[test_missing_tags_debian_feed](../doc/test_feeds/debian/test_missing_tags_debian_feed.md)**: Set of tests that check the behavior of Vulnerability Detector when any tag is missing from the feed.
```
